tasks/rh-sign-image/rh-sign-image.yaml

```diff
@@ -68,6 +68,21 @@ spec:
         N=$(params.concurrentLimit)  # The maximum number of images to be processed at once
         count=0
         COMPONENTS_LENGTH=$(jq '.components |length' ${SNAPSHOT_PATH})
+
+        function is_signed() {
+          local registry_reference=$1
+          local tag=$2
+          local manifest_digest=$3
+
+          # Check if the signature exists using skopeo
+          if skopeo inspect --raw "docker://${registry_reference}:${tag}" \
+            | jq -e ".signatures | any(.digest == \"${manifest_digest}\")"; then
```

Comment:
I think you should probably add a default value for the `.signature`. The problem is that if there are no signatures, `jq` will print an ugly error:
```sh
jq -e '.signatures | any(.digest == "aaa")' <<< '{}'
jq: error (at <stdin>:1): Cannot iterate over null (null)
```
You can fix it like this: `jq -e ".signatures // {} | any(.digest == \"${manifest_digest}\")"`

Reply:
Agree, fixed!